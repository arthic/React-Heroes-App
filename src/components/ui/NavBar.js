import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="nav">

            <Link
                className="nav-logo"
                to="/"
            >
                <h1>Heroes App</h1>
            </Link>

            <div className="nav-content">

                <NavLink
                    activeClassName="activo"
                    exact
                    to="/marvel"
                >
                    <p>Marvel</p>
                    <img
                        src="./assets/nav/marvel.png"
                        style={{maxWidth: 240}}
                        alt="marvel"
                    />
                </NavLink>

                <NavLink
                    activeClassName="activo"
                    exact
                    to="/dc"
                >
                    <p>DC</p>
                    <img
                        // src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/3d/DC_Comics_logo.svg/1200px-DC_Comics_logo.svg.png"
                        src="./assets/nav/dc.png"
                        style={{maxWidth: 140}}
                        alt="dc"
                    />
                </NavLink>
                <NavLink
                    activeClassName="activo"
                    exact
                    to="/search"
                >
                    <p>BÃºsqueda</p>
                    <div>
                        <i
                            className="fas fa-search"
                        >
                        </i>

                    </div>
                </NavLink>
            </div>
        </nav>
    )
}