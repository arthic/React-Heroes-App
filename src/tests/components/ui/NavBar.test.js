import React from 'react';
import '@testing-library/jest-dom'
import { mount } from 'enzyme';
import { NavBar } from '../../../components/ui/NavBar';
import { MemoryRouter, Router } from 'react-router-dom';

describe('Testing en <NavBar />', () => {
	const historyMock = {
		push: jest.fn(),
		replace: jest.fn(),
		location: {},
		listen: jest.fn(),
		createHref: jest.fn()
	}
	const wrapper = mount(
		<MemoryRouter>
			<Router history={historyMock}>
				<NavBar />
			</Router>
		</MemoryRouter>
	)

	test('Despliegue correcto', () => {
		expect(wrapper).toMatchSnapshot()
		expect(wrapper.find('.fas').exists()).toBe(true)
	})

});
