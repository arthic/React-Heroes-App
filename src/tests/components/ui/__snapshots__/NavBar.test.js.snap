// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing en <NavBar /> Despliegue correcto 1`] = `
<nav
  className="nav"
>
  <a
    className="nav-logo"
    onClick={[Function]}
  >
    <h1>
      Heroes App
    </h1>
  </a>
  <div
    className="nav-content"
  >
    <a
      aria-current={null}
      onClick={[Function]}
    >
      <p>
        Marvel
      </p>
      <img
        alt="marvel"
        src="./assets/nav/marvel.png"
        style={
          Object {
            "maxWidth": 240,
          }
        }
      />
    </a>
    <a
      aria-current={null}
      onClick={[Function]}
    >
      <p>
        DC
      </p>
      <img
        alt="dc"
        src="./assets/nav/dc.png"
        style={
          Object {
            "maxWidth": 140,
          }
        }
      />
    </a>
    <a
      aria-current={null}
      onClick={[Function]}
    >
      <p>
        BÃºsqueda
      </p>
      <div>
        <i
          className="fas fa-search"
        />
      </div>
    </a>
  </div>
</nav>
`;
